Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -422,3 +566,33 @@ def ReadFile(path):
         LOG_ACTION.log(LogType.Error, "Exception opening file " + path + " Error Code: " + str(error.errno) + " Error: " + error.message + error.strerror)
     return content
 
+def ConvertToUtf8(item):
+    return str(item).encode('utf-8')
+
+def GetWorkspaceListFromStateFile():
+    stateFileFullPath = os.path.join(PATH_NPM_STATE, FNAME_NPM_STATE)
+    fileContents = ReadFile(stateFileFullPath)
+    if fileContents is None:
+        return []
+    wsList = filter(IsUUID, fileContents.splitlines())
+    return map(ConvertToUtf8, wsList)
+
+def IsUUID(uuidStr):
+    try:
+        uuidOut = re.match('[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}\Z', uuidStr, re.I)
```

Comment:
To me, it seems like your code is trying to check if the directory name you've found is a workspace ID. In omsadmin.sh, we are trying to detect workspace ID format as well. For consistency, I think the regex you use here should match that.

I'm not an expert on what official format these take (I thought they were simply GUIDs).